app/uk/gov/hmrc/soletraderidentificationfrontend/services/RegistrationOrchestrationService.scala

```diff
@@ -30,12 +30,12 @@ class RegistrationOrchestrationService @Inject()(soleTraderIdentificationService
                                                  auditService: AuditService
                                                 )(implicit ec: ExecutionContext) {
 
-  def register(journeyId: String, regime: String)(implicit hc: HeaderCarrier): Future[RegistrationStatus] = for {
+  def register(journeyId: String, journeyConfig: JourneyConfig)(implicit hc: HeaderCarrier): Future[RegistrationStatus] = for {
     registrationStatus <- soleTraderIdentificationService.retrieveBusinessVerificationStatus(journeyId).flatMap {
       case Some(BusinessVerificationPass) | Some(SaEnrolled) => for {
         optNino <- soleTraderIdentificationService.retrieveNino(journeyId)
         optSautr <- soleTraderIdentificationService.retrieveSautr(journeyId)
-        registrationStatus <- register(journeyId, optNino, optSautr, regime)
+        registrationStatus <- register(journeyId, optNino, optSautr, journeyConfig.regime)
```

Comment:
I know you didn't write the original code, but a method called register calling a private method called register is very confusing. It might be worth renaming one these.

Reply:
Done